### PR TITLE
Feat/scout 21 regions module

### PR DESCRIPTION
**Pull request by dejwi**
[SCOUT-21](https://playmakerpro.atlassian.net/browse/SCOUT-21?atlOrigin=eyJpIjoiZGZjZDg1MTliYTFiNDBhN2JlODdhZDI1NWMxY2M3YjMiLCJwIjoiaiJ9)

### Task Description
Regions
- table with filter form
- create/edit form
- translations
- index/create/edit/id pages
- admin role check
- withSessionSsrRole helper

### Additional Notes (optional)
nie jest jeszcze w rozwijanej liście, w poprzednim pr zakładka admin panel została zmieniona na rozwijaną i nie wiem, ale reszta gotowa do zobaczenia

### CODE REVIEW

File: src/utils/withSessionSsrRole.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+export type TSsrRole<T = null> = {
+  errorStatus: number | null
+  errorMessage: string | null
+  data: T | null
+}
+type TRole = Components.Schemas.UserDto['role']
+
+export function withSessionSsrRole<T>(
+  _translations: string[],
```
Comment (jakubfiglak) [inline src/utils/withSessionSsrRole.ts:17]: Why did you prefix this variable name with an underscore?

File: src/utils/withSessionSsrRole.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  requiredRoles: TRole[],
```
Comment (jakubfiglak) [inline src/utils/withSessionSsrRole.ts:18]: I think `allowedRoles` would be a better name for this argument

File: src/utils/withSessionSsrRole.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+      locale || 'pl',
+      _translations,
+    )
+
+    if (!requiredRoles.includes(user.role)) {
+      return {
+        props: {
+          ...translations,
+          errorStatus: 401,
+          errorMessage: 'Insufficient Permissions',
```
Comment (jakubfiglak) [inline src/utils/withSessionSsrRole.ts:48]: I'd add this string to translations

File: src/utils/withSessionSsrRole.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+          data: null,
+        },
+      }
+    }
+
+    let data = null
+    if (getData) {
+      const resd = await getData(
```
Comment (jakubfiglak) [inline src/utils/withSessionSsrRole.ts:56]: Is it a typo or you wanted to name this variable `resd` with a `d` ?

**Review by jakubfiglak — APPROVED** [review]
🚀 Really good job, just some minor comments. Great idea with extracting this `withSessionSsrRole` helper. We might want to create another similar one to protect pages that require authentication (so basically most of them :D)

File: src/utils/withSessionSsrRole.ts (hunk 1 above)
Comment (dejwi) [reply src/utils/withSessionSsrRole.ts:17]: translations was taken but yeah I could've used another name😅

File: src/utils/withSessionSsrRole.ts (hunk 4 above)
Comment (dejwi) [reply src/utils/withSessionSsrRole.ts:56]: just a shortcut from resdata, can be changed if it's too much

**Pull request merged**